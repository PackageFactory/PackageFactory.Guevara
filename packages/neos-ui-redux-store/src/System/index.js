import {createAction} from 'redux-actions';

import {handleActions} from '@neos-project/utils-redux';

const BOOT = '@neos/neos-ui/System/BOOT';
const INIT = '@neos/neos-ui/System/INIT';
const READY = '@neos/neos-ui/System/READY';

//
// Export the action types
//
export const actionTypes = {
    BOOT,
    INIT,
    READY
};

const boot = createAction(BOOT);
const init = createAction(INIT, state => state);
const ready = createAction(READY);

//
// Export the actions
//
export const actions = {
    boot,
    init,
    ready
};

//
// Export the reducer
//
export const reducer = handleActions({
});

//
// Export the selectors
//
export const selectors = {};
