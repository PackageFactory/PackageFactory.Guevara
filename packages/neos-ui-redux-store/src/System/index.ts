import produce from 'immer';
import {$get} from 'plow-js';
import {action as createAction, ActionType as ActionTypeHelper} from 'typesafe-actions';

const BOOT = '@neos/neos-ui/System/BOOT';
const INIT = '@neos/neos-ui/System/INIT';
const READY = '@neos/neos-ui/System/READY';
const AUTHENTICATION_TIMEOUT = '@neos/neos-ui/System/AUTHENTICATION_TIMEOUT';
const REAUTHENTICATION_SUCCEEDED = '@neos/neos-ui/System/REAUTHENTICATION_SUCCEEDED';

//
// Export the action types
//
export const actionTypes = {
    AUTHENTICATION_TIMEOUT,
    BOOT,
    INIT,
    READY,
    REAUTHENTICATION_SUCCEEDED
};

interface State {
    readonly authenticationTimeout: boolean;
}

const boot = () => createAction(BOOT);
const init = (state: State) => createAction(INIT, state);
const ready = () => createAction(READY);
const authenticationTimeout = () => createAction(AUTHENTICATION_TIMEOUT);
const reauthenticationSucceeded = () => createAction(REAUTHENTICATION_SUCCEEDED);

//
// Export the actions
//
export const actions = {
    authenticationTimeout,
    boot,
    init,
    ready,
    reauthenticationSucceeded
};

export type ActionType = ActionTypeHelper<typeof actions>;

const defaultState: State = {
    authenticationTimeout: false
};

//
// Export the reducer
//
export const reducer = (state: State = defaultState, action: ActionType) => {
    return produce(state, draft => {
        switch (action.type) {
            case INIT:
                draft.authenticationTimeout = false;
                break;
            case AUTHENTICATION_TIMEOUT:
                draft.authenticationTimeout = true;
                break;
            case REAUTHENTICATION_SUCCEEDED:
                draft.authenticationTimeout = false;
                break;
        }
    });
};

//
// Export the selectors
//
export const selectors = {
    authenticationTimeout: (state: any) => $get(['system'], state).authenticationTimeout
};
