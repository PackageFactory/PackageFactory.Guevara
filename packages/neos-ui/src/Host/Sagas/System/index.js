import {take, put} from 'redux-saga/effects';
import {delay, discover} from 'Shared/Utilities/Promises';
import {actionTypes, actions} from 'Host/Redux/index';
import initializeJSAPI from 'API/index';

let injectStore = null;
export const getStore = discover(function * () {
    const store = yield new Promise(resolve => {
        injectStore = resolve;
    });

    return store;
});

export const getAppContainer = discover(function * () {
    const appContainer = yield new Promise(resolve => {
        document.addEventListener('DOMContentLoaded', () => {
            resolve(document.getElementById('appContainer'));
        });
    });

    return appContainer;
});

export const getCsrfToken = discover(function * () {
    const appContainer = yield getAppContainer;

    return appContainer.dataset.csrfToken;
});

export const getSystemEnv = discover(function * () {
    const appContainer = yield getAppContainer;

    return appContainer.dataset.env;
});

export const getServerState = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="initialState"]').innerHTML);
});

export const getConfiguration = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="configuration"]').innerHTML);
});

export const getMenu = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="menu"]').innerHTML);
});

export const getTranslations = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="translations"]').innerHTML);
});

export const getNeos = discover(function * () {
    const csrfToken = yield getCsrfToken;
    const systemEnv = yield getSystemEnv;

    const neos = initializeJSAPI(window, {
        csrfToken,
        systemEnv
    });

    return neos;
});

function * bootSaga(store) {
    injectStore(store);

    yield take(actionTypes.System.BOOT);

    const serverState = yield getServerState;

    //
    // Kick off initialization for redux, other sagas can
    // listen to this action and start their initialization tasks
    //
    yield put(actions.System.init(serverState));

    //
    // Just make sure that everybody does their initialization
    // homework
    //
    yield delay(0);

    //
    // Inform everybody, that we're ready to rock
    //
    yield put(actions.System.ready());
}

export const sagas = [
    bootSaga
];
