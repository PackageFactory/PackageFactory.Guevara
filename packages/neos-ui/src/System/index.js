import {discover} from '@neos-project/utils-helpers';
import {initializeJsAPI} from '@neos-project/neos-ui-backend-connector';

export const getAppContainer = discover(function * () {
    const appContainer = yield new Promise(resolve => {
        document.addEventListener('DOMContentLoaded', () => {
            resolve(document.getElementById('appContainer'));
        });
    });

    return appContainer;
});

export const getCsrfToken = discover(function * () {
    const appContainer = yield getAppContainer;

    return appContainer.dataset.csrfToken;
});

export const getSystemEnv = discover(function * () {
    const appContainer = yield getAppContainer;

    return appContainer.dataset.env;
});

export const getServerState = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="initialState"]').innerHTML);
});

export const getConfiguration = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="configuration"]').innerHTML);
});

export const getNodeTypes = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="nodeTypes"]').innerHTML);
});

export const getMenu = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="menu"]').innerHTML);
});

export const getTranslations = discover(function * () {
    const appContainer = yield getAppContainer;

    return JSON.parse(appContainer.querySelector('[data-json="translations"]').innerHTML);
});

export const getNeos = discover(function * () {
    const csrfToken = yield getCsrfToken;
    const systemEnv = yield getSystemEnv;

    const neos = initializeJsAPI(window, {
        csrfToken,
        systemEnv
    });

    return neos;
});
