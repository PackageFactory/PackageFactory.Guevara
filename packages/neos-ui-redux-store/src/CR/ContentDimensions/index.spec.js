import Immutable from 'immutable';
import {$all, $set, $get} from 'plow-js';

import {actionTypes, actions, reducer, selectors} from './index.js';
import {actionTypes as system} from '../../System/index';

const fixtures = {};

fixtures.dimensionState = $all(
    $set('cr.contentDimensions.active', {language: ['fr']}),
    $set('cr.contentDimensions.allowedPresets', {
        language: ['en_US', 'en_UK', 'de', 'fr']
    }),
    $set('cr.contentDimensions.byName', {
        language: {
            label: 'Language',
            icon: 'fa-language',
            default: 'en_US',
            defaultPreset: 'en_US',
            presets: {
                en_US: { // eslint-disable-line
                    label: 'English (US)',
                    values: ['en_US'],
                    uriSegment: 'en'
                },
                en_UK: { // eslint-disable-line
                    label: 'English (UK)',
                    values: ['en_UK'],
                    uriSegment: 'en-uk'
                },
                de: {
                    label: 'German',
                    values: ['de'],
                    uriSegment: 'de'
                },
                fr: {
                    label: 'French',
                    values: ['fr'],
                    uriSegment: 'fr'
                }
            }
        },
        topicfocus: {
            label: 'Topic focus',
            icon: 'fa-user',
            default: 'all',
            defaultPreset: 'all',
            presets: {
                all: {
                    label: 'All',
                    values: ['all'],
                    uriSegment: 'all'
                },
                family: {
                    label: 'Family',
                    values: ['family'],
                    uriSegment: 'family'
                },
                sports: {
                    label: 'Sports',
                    values: ['sports'],
                    uriSegment: 'sports'
                },
                adventure: {
                    label: 'Adventure',
                    values: ['adventure'],
                    uriSegment: 'adventure'
                }
            }
        }
    }),
    {}
);

test(`should export actionTypes`, () => {
    expect(actionTypes).not.toBe(undefined);
    expect(typeof (actionTypes.SELECT_PRESET)).toBe('string');
    expect(typeof (actionTypes.SET_ACTIVE)).toBe('string');
});

test(`should export action creators`, () => {
    expect(actions).not.toBe(undefined);
    expect(typeof (actions.selectPreset)).toBe('function');
    expect(typeof (actions.setActive)).toBe('function');
});

test(`should export a reducer`, () => {
    expect(reducer).not.toBe(undefined);
    expect(typeof (reducer)).toBe('function');
});

test(`should export selectors`, () => {
    expect(selectors).not.toBe(undefined);
});

test(`The reducer should create a valid initial state`, () => {
    const state = new Immutable.Map({});
    const nextState = reducer(state, {
        type: system.INIT,
        payload: fixtures.dimensionState
    });

    expect(nextState).toMatchSnapshot();
});

test(`SELECT_PRESET should add the selected preset's dimension values to the active state`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const action = actions.selectPreset({topicfocus: 'sports'});
    const nextState = reducer(state, action);

    expect(nextState).toMatchSnapshot();
});

test(`SELECT_PRESET should overwrite the currently selected dimension values`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const action = actions.selectPreset({language: 'en_UK'});
    const nextState = reducer(state, action);

    expect(nextState).toMatchSnapshot();
});

test(`SET_ACTIVE should overwrite the currently selected dimension values`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const action = actions.setActive({
        language: ['en_UK']
    });
    const nextState = reducer(state, action);

    expect(nextState).toMatchSnapshot();
});

test(`SET_ACTIVE should ignore dimensions that haven't been previously active`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const action = actions.setActive({
        language: ['en_UK'],
        topicfocus: ['sports']
    });
    const nextState = reducer(state, action);

    expect(nextState).toMatchSnapshot();
});

test(`allowedPresets selector should return the allowed Presets`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const allowedPresets = selectors.allowedPresets(state);

    expect(allowedPresets).toMatchSnapshot();
});

test(`active selector should return the active dimension values`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const active = selectors.active(state);

    expect(active).toMatchSnapshot();
});

test(`activePresets selector should return the active dimensions configuration`, () => {
    const state = Immutable.fromJS(fixtures.dimensionState);
    const activePresets = selectors.activePresets(state);

    expect(activePresets).toMatchSnapshot();
});

test(`activePresets selector should fall back to default preset, if none is set`, () => {
    const state = Immutable.fromJS(
        $set('cr.contentDimensions.active.language', null, fixtures.dimensionState)
    );
    const activePresets = selectors.activePresets(state);

    expect($get('language.name', activePresets)).toBe('en_US');
});
