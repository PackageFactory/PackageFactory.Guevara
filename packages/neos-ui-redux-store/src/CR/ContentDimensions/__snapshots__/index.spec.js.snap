// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SELECT_PRESET should add the selected preset's dimension values to the active state 1`] = `
Immutable.Map {
  "cr": Immutable.Map {
    "contentDimensions": Immutable.Map {
      "active": Immutable.Map {
        "language": Immutable.List [
          "fr",
        ],
        "topicfocus": Immutable.List [
          "sports",
        ],
      },
      "allowedPresets": Immutable.Map {
        "language": Immutable.List [
          "en_US",
          "en_UK",
          "de",
          "fr",
        ],
      },
      "byName": Immutable.Map {
        "language": Immutable.Map {
          "label": "Language",
          "icon": "fa-language",
          "default": "en_US",
          "defaultPreset": "en_US",
          "presets": Immutable.Map {
            "en_US": Immutable.Map {
              "label": "English (US)",
              "values": Immutable.List [
                "en_US",
              ],
              "uriSegment": "en",
            },
            "en_UK": Immutable.Map {
              "label": "English (UK)",
              "values": Immutable.List [
                "en_UK",
              ],
              "uriSegment": "en-uk",
            },
            "de": Immutable.Map {
              "label": "German",
              "values": Immutable.List [
                "de",
              ],
              "uriSegment": "de",
            },
            "fr": Immutable.Map {
              "label": "French",
              "values": Immutable.List [
                "fr",
              ],
              "uriSegment": "fr",
            },
          },
        },
        "topicfocus": Immutable.Map {
          "label": "Topic focus",
          "icon": "fa-user",
          "default": "all",
          "defaultPreset": "all",
          "presets": Immutable.Map {
            "all": Immutable.Map {
              "label": "All",
              "values": Immutable.List [
                "all",
              ],
              "uriSegment": "all",
            },
            "family": Immutable.Map {
              "label": "Family",
              "values": Immutable.List [
                "family",
              ],
              "uriSegment": "family",
            },
            "sports": Immutable.Map {
              "label": "Sports",
              "values": Immutable.List [
                "sports",
              ],
              "uriSegment": "sports",
            },
            "adventure": Immutable.Map {
              "label": "Adventure",
              "values": Immutable.List [
                "adventure",
              ],
              "uriSegment": "adventure",
            },
          },
        },
      },
    },
  },
}
`;

exports[`SELECT_PRESET should overwrite the currently selected dimension values 1`] = `
Immutable.Map {
  "cr": Immutable.Map {
    "contentDimensions": Immutable.Map {
      "active": Immutable.Map {
        "language": Immutable.List [
          "en_UK",
        ],
      },
      "allowedPresets": Immutable.Map {
        "language": Immutable.List [
          "en_US",
          "en_UK",
          "de",
          "fr",
        ],
      },
      "byName": Immutable.Map {
        "language": Immutable.Map {
          "label": "Language",
          "icon": "fa-language",
          "default": "en_US",
          "defaultPreset": "en_US",
          "presets": Immutable.Map {
            "en_US": Immutable.Map {
              "label": "English (US)",
              "values": Immutable.List [
                "en_US",
              ],
              "uriSegment": "en",
            },
            "en_UK": Immutable.Map {
              "label": "English (UK)",
              "values": Immutable.List [
                "en_UK",
              ],
              "uriSegment": "en-uk",
            },
            "de": Immutable.Map {
              "label": "German",
              "values": Immutable.List [
                "de",
              ],
              "uriSegment": "de",
            },
            "fr": Immutable.Map {
              "label": "French",
              "values": Immutable.List [
                "fr",
              ],
              "uriSegment": "fr",
            },
          },
        },
        "topicfocus": Immutable.Map {
          "label": "Topic focus",
          "icon": "fa-user",
          "default": "all",
          "defaultPreset": "all",
          "presets": Immutable.Map {
            "all": Immutable.Map {
              "label": "All",
              "values": Immutable.List [
                "all",
              ],
              "uriSegment": "all",
            },
            "family": Immutable.Map {
              "label": "Family",
              "values": Immutable.List [
                "family",
              ],
              "uriSegment": "family",
            },
            "sports": Immutable.Map {
              "label": "Sports",
              "values": Immutable.List [
                "sports",
              ],
              "uriSegment": "sports",
            },
            "adventure": Immutable.Map {
              "label": "Adventure",
              "values": Immutable.List [
                "adventure",
              ],
              "uriSegment": "adventure",
            },
          },
        },
      },
    },
  },
}
`;

exports[`SET_ACTIVE should ignore dimensions that haven't been previously active 1`] = `
Immutable.Map {
  "cr": Immutable.Map {
    "contentDimensions": Immutable.Map {
      "active": Immutable.Map {
        "language": Immutable.List [
          "en_UK",
        ],
      },
      "allowedPresets": Immutable.Map {
        "language": Immutable.List [
          "en_US",
          "en_UK",
          "de",
          "fr",
        ],
      },
      "byName": Immutable.Map {
        "language": Immutable.Map {
          "label": "Language",
          "icon": "fa-language",
          "default": "en_US",
          "defaultPreset": "en_US",
          "presets": Immutable.Map {
            "en_US": Immutable.Map {
              "label": "English (US)",
              "values": Immutable.List [
                "en_US",
              ],
              "uriSegment": "en",
            },
            "en_UK": Immutable.Map {
              "label": "English (UK)",
              "values": Immutable.List [
                "en_UK",
              ],
              "uriSegment": "en-uk",
            },
            "de": Immutable.Map {
              "label": "German",
              "values": Immutable.List [
                "de",
              ],
              "uriSegment": "de",
            },
            "fr": Immutable.Map {
              "label": "French",
              "values": Immutable.List [
                "fr",
              ],
              "uriSegment": "fr",
            },
          },
        },
        "topicfocus": Immutable.Map {
          "label": "Topic focus",
          "icon": "fa-user",
          "default": "all",
          "defaultPreset": "all",
          "presets": Immutable.Map {
            "all": Immutable.Map {
              "label": "All",
              "values": Immutable.List [
                "all",
              ],
              "uriSegment": "all",
            },
            "family": Immutable.Map {
              "label": "Family",
              "values": Immutable.List [
                "family",
              ],
              "uriSegment": "family",
            },
            "sports": Immutable.Map {
              "label": "Sports",
              "values": Immutable.List [
                "sports",
              ],
              "uriSegment": "sports",
            },
            "adventure": Immutable.Map {
              "label": "Adventure",
              "values": Immutable.List [
                "adventure",
              ],
              "uriSegment": "adventure",
            },
          },
        },
      },
    },
  },
}
`;

exports[`SET_ACTIVE should overwrite the currently selected dimension values 1`] = `
Immutable.Map {
  "cr": Immutable.Map {
    "contentDimensions": Immutable.Map {
      "active": Immutable.Map {
        "language": Immutable.List [
          "en_UK",
        ],
      },
      "allowedPresets": Immutable.Map {
        "language": Immutable.List [
          "en_US",
          "en_UK",
          "de",
          "fr",
        ],
      },
      "byName": Immutable.Map {
        "language": Immutable.Map {
          "label": "Language",
          "icon": "fa-language",
          "default": "en_US",
          "defaultPreset": "en_US",
          "presets": Immutable.Map {
            "en_US": Immutable.Map {
              "label": "English (US)",
              "values": Immutable.List [
                "en_US",
              ],
              "uriSegment": "en",
            },
            "en_UK": Immutable.Map {
              "label": "English (UK)",
              "values": Immutable.List [
                "en_UK",
              ],
              "uriSegment": "en-uk",
            },
            "de": Immutable.Map {
              "label": "German",
              "values": Immutable.List [
                "de",
              ],
              "uriSegment": "de",
            },
            "fr": Immutable.Map {
              "label": "French",
              "values": Immutable.List [
                "fr",
              ],
              "uriSegment": "fr",
            },
          },
        },
        "topicfocus": Immutable.Map {
          "label": "Topic focus",
          "icon": "fa-user",
          "default": "all",
          "defaultPreset": "all",
          "presets": Immutable.Map {
            "all": Immutable.Map {
              "label": "All",
              "values": Immutable.List [
                "all",
              ],
              "uriSegment": "all",
            },
            "family": Immutable.Map {
              "label": "Family",
              "values": Immutable.List [
                "family",
              ],
              "uriSegment": "family",
            },
            "sports": Immutable.Map {
              "label": "Sports",
              "values": Immutable.List [
                "sports",
              ],
              "uriSegment": "sports",
            },
            "adventure": Immutable.Map {
              "label": "Adventure",
              "values": Immutable.List [
                "adventure",
              ],
              "uriSegment": "adventure",
            },
          },
        },
      },
    },
  },
}
`;

exports[`The reducer should create a valid initial state 1`] = `
Immutable.Map {
  "cr": Immutable.Map {
    "contentDimensions": Immutable.Map {
      "byName": Immutable.OrderedMap {
        "language": Immutable.OrderedMap {
          "label": "Language",
          "icon": "fa-language",
          "default": "en_US",
          "defaultPreset": "en_US",
          "presets": Immutable.OrderedMap {
            "en_US": Immutable.OrderedMap {
              "label": "English (US)",
              "values": Immutable.List [
                "en_US",
              ],
              "uriSegment": "en",
            },
            "en_UK": Immutable.OrderedMap {
              "label": "English (UK)",
              "values": Immutable.List [
                "en_UK",
              ],
              "uriSegment": "en-uk",
            },
            "de": Immutable.OrderedMap {
              "label": "German",
              "values": Immutable.List [
                "de",
              ],
              "uriSegment": "de",
            },
            "fr": Immutable.OrderedMap {
              "label": "French",
              "values": Immutable.List [
                "fr",
              ],
              "uriSegment": "fr",
            },
          },
        },
        "topicfocus": Immutable.OrderedMap {
          "label": "Topic focus",
          "icon": "fa-user",
          "default": "all",
          "defaultPreset": "all",
          "presets": Immutable.OrderedMap {
            "all": Immutable.OrderedMap {
              "label": "All",
              "values": Immutable.List [
                "all",
              ],
              "uriSegment": "all",
            },
            "family": Immutable.OrderedMap {
              "label": "Family",
              "values": Immutable.List [
                "family",
              ],
              "uriSegment": "family",
            },
            "sports": Immutable.OrderedMap {
              "label": "Sports",
              "values": Immutable.List [
                "sports",
              ],
              "uriSegment": "sports",
            },
            "adventure": Immutable.OrderedMap {
              "label": "Adventure",
              "values": Immutable.List [
                "adventure",
              ],
              "uriSegment": "adventure",
            },
          },
        },
      },
      "active": Immutable.OrderedMap {
        "language": Immutable.List [
          "fr",
        ],
      },
      "allowedPresets": Immutable.OrderedMap {
        "language": Immutable.List [
          "en_US",
          "en_UK",
          "de",
          "fr",
        ],
      },
    },
  },
}
`;

exports[`active selector should return the active dimension values 1`] = `
Immutable.Map {
  "language": Immutable.List [
    "fr",
  ],
}
`;

exports[`activePresets selector should return the active dimensions configuration 1`] = `
Immutable.Map {
  "language": Immutable.Map {
    "label": "French",
    "values": Immutable.List [
      "fr",
    ],
    "uriSegment": "fr",
    "name": "fr",
  },
}
`;

exports[`allowedPresets selector should return the allowed Presets 1`] = `
Immutable.Map {
  "language": Immutable.List [
    "en_US",
    "en_UK",
    "de",
    "fr",
  ],
}
`;
